contracts/libraries/TokenAuthorized.sol

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.20;
+
+library TokenAuthorized {
```

Comment:
would love a comment above this! just like 1-2 lines explaining the purpose of the library